src/main/content/_assets/js/instance-common.js

```diff
@@ -60,6 +59,49 @@ function fetchATool(tool){
         .catch(error => console.error(`Error getting ${tool} tool`, error));
 }
 
+function fetchCollectionData(instanceName){
+    if(typeof instanceName === "undefined"){
+        return;
+    }
+
+    return fetch(`/api/kabanero/${instanceName}/collections`)
+        .then(function(response) {
+            return response.json();
+        })
+        .catch(error => console.error("Error getting collections", error));
+}
+
+function handleInitialCLIAuth(instanceName){
+    return fetch(`/api/auth/kabanero/${instanceName}/collections/list`)
+        .then(function(response) {
+
+            // Login via cli and retry if 401 is returned on initial call
+            if(response.status === 401){
+                return loginViaCLI(instanceName)
+                    .then(() => {
+                        return handleInitialCLIAuth(instanceName); 
+                    });
+            }
+            else if(response.status !== 200){
+                console.warn(`Initial auth into instance ${instanceName} returned status code: ${response.status}`);
+            }
+
+            // pass on instance name var to the next function in the promise chain
+            return instanceName;
+        })
+        .catch(error => console.error(`Error handling initial auth into instance ${instanceName} via CLI server`, error));
+}
+
+function loginViaCLI(instanceName){
+    if(typeof instanceName === "undefined"){
+        console.warn("CLI login cannot login without an instanceName");
+        return;
+    }
+
+    return fetch(`/api/auth/kabanero/${instanceName}/collections/login`)
+        .catch(error => console.error(`Error logging into instance ${instanceName} via CLI server`, error));
+}
+
```

Comment:
these 3 functions are already in `src/main/content/_assets/js/collections.js` so this is duplication, any reason you added them here too?